### ISSUE TITLE
GitHub action for automated documentation

### ISSUE THREAD

**Issue by taylorkaren** [department-of-veterans-affairs/va-forms-system-core#175]
## Description
The config is very simple right now and allows for us to configure a custom domain at a later time. This also allows us to change the theme/look of the documentation (which i recommend one of the built ins and not a custom plugin theme that is being used now).

Need to figure out a way in the _config.yml file or somewhere else to run the `yarn docs:build` command so that when we check push to master it will auto build the latest changes. 
- This might be a good thing to test, if you update actual content in the pages will it auto update? 

## Considerations
- There is already an auto generated github action to publish the site
- This action reads off of the _config.yml file located under the `docs` directory
- Github pages settings are located on Github under: Settings -> Pages 

## Tasks
- [x] Update workflow to include building of new doc references IF, it DOES NOT do this already
- [x] Update the repo readme file (https://github.com/department-of-veterans-affairs/va-forms-system-core/blob/master/README.md) to reference the new documentation location.

## Acceptance Criteria
- [x] When creating a new markdown document in the `/docs` directory, the page gets automatically updated and pushed to the live site.
- [x] Repo readme file (https://github.com/department-of-veterans-affairs/va-forms-system-core/blob/master/README.md) is updated to reference the new documentation location.

**Comment by mlauren** [comment]
@bjmccotter7192 I went ahead and assigned this to myself bc I was interested in this / have done very minimal gh actions integrations before. However I have a lot of questions about what is to be automated, would this be a container that pulls in confluence documentation that we have currently written? We can talk about this later

**Comment by bjmccotter7192** [comment]
word this service generates it's own docs using a yarn command of `yarn docs:build` this bundles and builds the markdown files and builds their references. This is then being hosted on github pages when you check in to the master branch.

**Comment by mlauren** [comment]
@bjmccotter7192 is there already a repo that we would be adding an action to? Do you think this would warrant creating one for the fl team?

**Comment by bjmccotter7192** [comment]
this would be included in the "va-forms-system-core" repo and add a new workflow or update one of the existing.

**Comment by bjmccotter7192** [comment]
Leaving the automated github action for github page deploy because it is already generating the site correctly.

Decided to leave the theme and just fix the title because other themes would require changes to HTML files and creating separate files not included in the api-extractor package.

Tried the dark theme (i personally like it) but some links were too dark and I thought it might not be the best for everyone.

Leaving the link in the README since the location did not change and is already working as intended.

**UPDATE MADE** - I did change the NPM publish workflow to only run after the Build and Test workflow completes successfully, also changed the "build" step to include the docs:build command. 

All that is required before check in is `yarn install` and `yarn build`. As long as there are no errors in either of those commands the docs will show as expected when pushed to master.

**Comment by bjmccotter7192** [comment]
Waiting on support to get back with me so we can swap out the NPM token with a production token

**Comment by bjmccotter7192** [comment]
IT Support is working to create a AWS service account for our team, this may take some time so I don't think we need to hold on this ticket because of that account. It is not needed for this ticket to be completed.

**Comment by taylorkaren** [comment]
@sdk-irl to review the PR.
